pkg/config/analysis/incluster/controller.go

```diff
@@ -78,27 +80,33 @@ func NewController(stop <-chan struct{}, rwConfigStore model.ConfigStoreControll
 
 // Run is blocking
 func (c *Controller) Run(stop <-chan struct{}) {
-	t := time.NewTicker(features.AnalysisInterval)
-	oldmsgs := diag.Messages{}
-	for {
-		select {
-		case <-t.C:
-			res, err := c.analyzer.ReAnalyze(stop)
-			if err != nil {
-				log.Errorf("In-cluster analysis has failed: %s", err)
-				continue
-			}
-			// reorganize messages to map
-			index := map[status.Resource]diag.Messages{}
-			for _, m := range res.Messages {
-				key := status.ResourceFromMetadata(m.Resource.Metadata)
-				index[key] = append(index[key], m)
-			}
-			// if we previously had a message that has been removed, ensure it is removed
-			// TODO: this creates a state destruction problem when istiod crashes
-			// in that old messages may not be removed.  Not sure how to fix this
-			// other than write every object's status every loop.
-			for _, m := range oldmsgs {
+	db := concurrent.Debouncer[config.GroupVersionKind]{}
+	chKind := make(chan config.GroupVersionKind, 10)
```

Comment:
Actually I may have read it wrong, forgot we do `go push(...)`, so it shouldn't block (I mean, it can -- but its doing trivial work, not the full analysis). So worth a 2nd look but could be fine